packages/castmill/lib/castmill/addons/playlists/components/index.tsx

```diff
@@ -200,8 +204,190 @@ const PlaylistsPage: Component<AddonComponentProps> = (props) => {
     setShowModal(false);
   };
 
+  const [customRatioModal, setCustomRatioModal] = createSignal<{
+    playlist: JsonPlaylist;
+    show: boolean;
+  }>({ playlist: null as any, show: false });
+
+  const [customWidth, setCustomWidth] = createSignal<string>('16');
+  const [customHeight, setCustomHeight] = createSignal<string>('9');
+  const [customRatioErrors, setCustomRatioErrors] = createSignal(
+    new Map<string, string>()
+  );
+
+  const validateCustomRatio = (field: string, value: string) => {
+    const errors = new Map(customRatioErrors());
+    const num = parseInt(value, 10);
+    if (!value || isNaN(num)) {
+      errors.set(field, t('playlists.errors.aspectRatioNumber'));
+    } else if (num <= 0) {
+      errors.set(field, t('playlists.errors.aspectRatioPositive'));
+    } else if (num > MAX_ASPECT_RATIO_VALUE) {
+      errors.set(field, t('playlists.errors.aspectRatioMax'));
+    } else {
+      errors.delete(field);
+    }
+    setCustomRatioErrors(errors);
+    return !errors.has(field);
+  };
+
+  const validateCustomRatioExtreme = () => {
+    const width = parseInt(customWidth(), 10);
+    const height = parseInt(customHeight(), 10);
+
+    if (!isNaN(width) && !isNaN(height) && width > 0 && height > 0) {
+      const ratio = width / height;
+      if (ratio > MAX_ASPECT_RATIO || ratio < MIN_ASPECT_RATIO) {
+        const errors = new Map(customRatioErrors());
+        errors.set('ratio', t('playlists.errors.aspectRatioExtreme'));
+        setCustomRatioErrors(errors);
+        return false;
+      } else {
+        const errors = new Map(customRatioErrors());
+        errors.delete('ratio');
+        setCustomRatioErrors(errors);
+      }
+    }
+    return true;
+  };
+
+  const handleCustomRatioSubmit = async () => {
+    if (
+      !validateCustomRatio('width', customWidth()) ||
+      !validateCustomRatio('height', customHeight()) ||
+      !validateCustomRatioExtreme()
+    ) {
+      return;
+    }
+
+    const width = parseInt(customWidth(), 10);
+    const height = parseInt(customHeight(), 10);
+    const playlist = customRatioModal().playlist;
+
+    try {
+      await PlaylistsService.updatePlaylist(
+        props.store.env.baseUrl,
+        props.store.organizations.selectedId,
+        `${playlist.id}`,
+        {
+          name: playlist.name,
+          description: '',
+          settings: {
+            aspect_ratio: { width, height },
+          },
+        }
+      );
+
+      toast.success(t('playlists.aspectRatioUpdated'));
+      setCustomRatioModal({ playlist: null as any, show: false });
```

Comment:
maybe allow null as a valid type for playlist if we need than instead of using any